### ISSUE TITLE
Implement contract management page

### ISSUE THREAD

**Issue by amenity** [cityofaustin/atd-data-tech#9857]
SMB SharePoint Contract [list](https://cityofaustin.sharepoint.com/:x:/r/sites/atdSMB/Shared%20Documents/Contracts/SMD_ContractTracker.xlsx?d=w0b961cfe4ef94af6899184d1035d5564&csf=1&web=1&e=zsy0rP)

## Fields currently tracking:
- DCD
- DDEP
- DOCUMENT ID
- DOCUMENT DESC
- AUTHORIZED-AMT
- ENCUMBRANCE-AMT
- EXPENDITURES-AMT
- BALANCE-AMT
- PCT AMT
- PCT TIME
- VENDOR CODE
- VENDOR NAME
- BEGIN DATE
- EXPIR DATE
- OPT-CNT
- LEN
- UNIT
- COMMENTS
- PRICE INCREASE
- LEAD TIME COMMENTS
- CONTACTS

![image](https://user-images.githubusercontent.com/21112975/192010976-eec688e1-041c-4fcb-92d3-75ce0e56bad0.png)
![image](https://user-images.githubusercontent.com/21112975/192011059-53a57ab0-2e5e-49e8-b875-7710a3748951.png)

**Comment by amenity** [comment]
@maccallump may have data sources to help with this.

**Comment by dianamartin** [comment]
Looked at eCapri 
Seems to have a page for **Contracts** [page](https://ecapris.austintexas.gov/index.cfm?fuseaction=contractsSrch.searchResults)
- AMD and SMB have a need to pull this information, wonder if it should live on the Finance and Purchasing app?

This is the same source (eCapri) as the Tasks table that is currently in AMD. I think Charlie/John would be able to tap into this to have an integration.

**Comment by dianamartin** [comment]
Resources: 
- [Real-Time M.A. Tabbed Spreadsheet Query](http://purchweb.austintexas.gov/search/RTMATSSwitchboard.cfm)
- [Search For Master Agreements](http://purchweb.austintexas.gov/search/SVMA.cfm)

**Comment by patrickm02L** [comment]
@dianamartin can you update what John said about the Master agreement agreement?

Then follow up with finance to get clarity on how to work on their app.

**Comment by dianamartin** [comment]
I had asked John about the `master_agreement` object in the Finance and Purchasing app if it was being updated automatically, he said
>yes—they are updated by legacy [script](https://deploy-preview-397--austin-transportation.netlify.app/publisher)

**Comment by dianamartin** [comment]
@dianamartin Ask @johnclary about what the plan is for Contract Management in MOPED, don't want duplicative solutions going on.

**Comment by johnclary** [comment]
@dianamartin here's what our contract tracking table looks like. We're not doing any any validation of contract #'s or vendors. It's free text right now.

![contracts](https://user-images.githubusercontent.com/14793120/183696375-32004b68-5e25-4178-950e-78e543f206a6.png)

**Comment by dianamartin** [comment]
@johnclary is it being piped in from eCapri? Advantage3 system?

**Comment by amenity** [comment]
@johnclary - I told @dianamartin and SMBers that these aren't currently coming straight from eCapris or Advantage3. 

Kati wasn't in attendance, but we think her biggest pain point is the inability to get 
- additional expiration notifications -  current ones are 1 month in advance, she'd like 6 
- price changes on (mostly on material)
- having it as a reference for inventory personnel
- used amount & outstanding balance 

PS Christina loves the idea of everyone aligning on a contract management system - AMD asked for this too, right?

**Comment by dianamartin** [comment]
# 9/23
Shared the update with the SMB stakeholders that this might actually be happening, but in the Finance and Purchasing application rather than the SMB tracker.

**Comment by ChristinaTremel** [comment]
### 1/27
Diana has a meeting with finance next Tuesday to talk about recently published test page. Finance wanted TKs to show up in the finance system from AIMS. That is the start for a place where we can centralize information. Also have the ability to bring in contract records, but just haven't yet.

**Comment by patrickm02L** [comment]
Apps teams needs to create a contract management table, then have a Dev (Charlie) pull the data from the data warehouse and pull it into the table. 

- similar to #10960. 
- The object ID for `financial_contracts` is @dianamartin

**Comment by dianamartin** [comment]
# 3/29

- [x] Created object `financial_contracts` in Finance and Purchasing
- [ ] Unsure of the field headings that are available in the Data Warehouse @Charlie-Henry are you able to check for me?

# Option 1
This [page](http://purchweb.austintexas.gov/search/RTMATSReport.cfm)

## Field Headings

- DCD
- DDEP
- DOCUMENT ID
- DOCUMENT DESCRIPTION
- AWARD CODE
- ADEP
- AUTHORIZED-AMT
- ENCUMBRANCES-AMT
- EXPENDITURES-AMT
- BALANCE-AMT
- PCT AMT
- PCT TIME
- FLAG
- BUYER
- REQUESTOR
- VEND-CNT
- VENDOR CODE
- VENDOR NAME
- BEGIN DATE
- EXPIR DATE
- OPT-CNT
- LEN
- UNIT
- RPT-1
- GO CAT

Other [page](http://purchweb.austintexas.gov/search/DispVendMA.cfm?vend_cust_cd=&vennam=&venalias=&vencode=&dept=2400&authdept=&document_id=&status=1&description=&buyer=&end_date=&end_date_from=&end_date_to=&display_vendor=2&comm_list=)

![image](https://user-images.githubusercontent.com/21112975/228676271-0ce484f3-8dd9-40ca-bb3a-1b54c19c34dc.png)

## Field Headings
- Code 
- ID 
- Dept 
- End Date 
- Total Orders 
- Authorized Amount 
- Remaining Balance 
- Description

**Comment by Charlie-Henry** [comment]
@dianamartin can you refresh me on what we call the "data warehouse"? 

Is it the same [database we are pulling TKs from](https://github.com/cityofaustin/atd-data-tech/issues/10111)?

**Comment by dianamartin** [comment]
@Charlie-Henry Yep, you got it. I should have probably included that piece of information 😉

**Comment by Charlie-Henry** [comment]
Table `DEPT_2400_MA_VW` sounds right?

Here's one record:
```
  {
    "DOC_CD": "MA",
    "DOC_DEPT_CD": "2400",
    "DOC_ID": "GA110000045",
    "DOC_DSCR": "UPS For Traffic Signals",
    "DOC_PHASE_CD": 3,
    "VEND_CUST_CD": "REP8307191",
    "LGL_NM": "REPUBLIC INTELLIGENT TRANSPORTATION SERVICES INC"
  }
```

**Comment by dianamartin** [comment]
@Charlie-Henry that looks right to me. Can you see all the fields that are available? I'll need to check them against the data the Signs and Markings currently tracks.

**Comment by dianamartin** [comment]
# 3/30
Charlie and I discussing the table headings
Charlie said
>the database we pull data from only has those 7 columns, I thought you were asking for what was possible we could bring in?

I said essentially, that those 7 columns don't include "Expiration Date", "Balance Amount", etc..

Charlie said
>if they could just add more columns to that view that would be easiest

I said we did do this for the TK data feed for #10111

## Emails
### Email 1 - to Charlie & Patrick
- I found the email that I reached out to FSD IT, forward to Charlie and Patrick
>This is the email that I had to update the existing TK Financial View (Data Warehouse).

>(to Patrick) We are thinking we might need to update the existing Contract View (Data Warehouse), unless they have a better table for us to tap into. Charlie says we only on have 7 fields available right now and we’re missing several that stakeholders would like to see. Charlie said there is an existing MA (contracts) microstrategy report that we can maybe use instead.

### Email 2 - to FSD IT
Email to Kai, cc: Bruce, Patrick (FSD IT)
>Hi Kai, 
Who would we communicate with to find out more the table DEPT_2400_MA_VW table? We may have a request to add more columns to this table. How would we go about requesting this when we’re ready? 
>Also, we’re building another ETL to use this table DEPT_2400_MA_VW, but should we use an existing “Master Agreement” Microstrategy report that we discovered instead to use as the basis of our data feed?

**Comment by johnclary** [comment]
nice work y'all! my two cents is that it would be nice to build on the existing DB view and incorporate the data into the [atd-finance-data pipeline](https://github.com/cityofaustin/atd-finance-data)

**Comment by dianamartin** [comment]
@johnclary Yeah, I agree with this route rather than create a new ETL using MicroStrategy data.

**Comment by Charlie-Henry** [comment]
I agree @dianamartin @johnclary, fastest would definitely be just adding another table to atd-finance-data. We do have existing ETLs to get microstrategy reports but it's for bond reporting in a separate repo.

**Comment by johnclary** [comment]
cool sounds good!

**Comment by dianamartin** [comment]
# 6/13
![image](https://github.com/cityofaustin/atd-data-tech/assets/21112975/bd56a4cf-036c-45ee-b515-013494c8accb)

![image](https://github.com/cityofaustin/atd-data-tech/assets/21112975/608fe6ba-d91d-48d4-94b3-39f7814bdbd9)

![image](https://github.com/cityofaustin/atd-data-tech/assets/21112975/9e3d2270-721a-4a8a-ad0e-84eb29692946)
